src/ucp/core/ucp_rkey.c

```diff
@@ -34,6 +34,26 @@ size_t ucp_rkey_packed_size(ucp_context_h context, ucp_md_map_t md_map)
     return size;
 }
 
+void ucp_rkey_packed_copy(ucp_context_h context, ucp_md_map_t md_map,
+                          void *rkey_buffer, const void* uct_rkeys[])
+{
+    void *p = rkey_buffer;
+    unsigned md_index;
+    size_t md_size;
+
+    *(ucp_md_map_t*)p = md_map;
+    p += sizeof(ucp_md_map_t);
+
+    ucs_for_each_bit(md_index, md_map) {
+        md_size = context->tl_mds[md_index].attr.rkey_packed_size;
+        ucs_assert_always(md_size <= UINT8_MAX);
+        *((uint8_t*)p++) = md_size;
+        memcpy(p, *uct_rkeys, md_size);
+        p += sizeof(uint8_t) + md_size;
```

Comment:
seems like need to do `p += md_size`, because you already made `p++` before

Reply:
right, it worked because offload currently has only 1 key